src/pk-engine.c

```diff
@@ -1755,6 +1755,38 @@ pk_engine_offline_method_call (GDBusConnection *connection_, const gchar *sender
 		g_dbus_method_invocation_return_value (invocation, value);
 		return;
 	}
+	if (g_strcmp0 (method_name, "GetResults") == 0) {
+		g_autoptr(PkResults) results = NULL;
+		gboolean success = FALSE;
+		g_autoptr(PkError) pk_error = NULL;
+		const gchar *err_descr = "";
+		GVariant *value = NULL;
+
+		results = pk_offline_get_results (&error);
+		if (results == NULL || pk_results_get_exit_code (results) != PK_EXIT_ENUM_SUCCESS) {
+			g_dbus_method_invocation_return_error (invocation,
+							       PK_ENGINE_ERROR,
+							       PK_ENGINE_ERROR_INVALID_STATE,
+							       "%s", error->message);
+			return;
+		}
+
+		pk_error = pk_results_get_error_code (results);
```

Comment:
Check if "success" is set as below, and if "success" is FALSE, do this:

```c
PkErrorEnum error_code = PK_ERROR_ENUM_UNKNOWN;
g_autofree gchar *error_msg = NULL;
g_autoptr(PkError) pk_error = NULL;
pk_error = pk_results_get_error_code (results);
if (pk_error == NULL) {
        error_code = PK_ERROR_ENUM_UNKNOWN
	error_msg = g_strdup("Offline update failed without error_code set");
} else {
        error_code = pk_error_get_code (pk_error),
	error_msg = g_strdup( pk_error_get_details (error_code));
}
```
And then return `success`, `error_code` and `error_msg` from the DBus call. This will make sure we handle even odd error corner cases.